### ISSUE TITLE
24-bit Address support

### ISSUE THREAD

**Issue by thetechknight** [nkaminski/memsim_68k#1]
Just saw this library, will make it easier to debug an unknown proprietary system, but its lacking 24-bit addressing support for a full 68K

**Comment by nkaminski** [comment]
Debugging an unknown proprietary system...... sure sounds interesting ;)

It shouldn't be difficult to expand the simulated address bus to be 24 bits wide, however I do not have an m68k peripheral to test against, so we might need to develop/test this collaboratively. The first touch points that I can think of will be the initialization routines, where we will need to configure another PORT/DDR register pair, as well as initialize all of PORTG/DDRG to set all of such as output. The set_address function will also need to be expanded to set 3, 8 bit IO PORT registers for a full 24 bit address.

**Comment by thetechknight** [comment]
Well its gonna be a fun one to debug because the software that ran it was downloaded over satellite and that signal has been gone for years. I would like to get it running for fun/prosperity reasons as I respect its vintage hardware. It basically has a Ready Systems RTOS ROM which no documentation exists for (it was a proprietary system). So unless you are really good at disassembly (I am not) then I cant use it. 

But anyways, this monster rocks TWO 68010 CPUs. One is on the main CPU card, contains the DRAM as well as the RTC chip, and the DMA/VMEBus logic. the 2nd card whcih I am more interested in, is the graphics card. it also has a 68010 and also contains an 8051 used to run the sprite/roll/craw/etc. along with 2MB of its own DRAM. the 68010 renders graphics primitives (which was also downloaded) into the VRAM, and the FPGAs (XC2018) control and buffer access from the 68K and the pixel clock readout. along with controlling sprite animations, etc. the output of the system uses a PS/2 style RAMDAC for 256 color support, and the board is genlocking. So it has an NTSC encoder that can mix the signals from the graphics and video. so it does overlays on the LDL, as well as full screen graphics when needed. Kind of a fun little card! 

The graphics board also has a ROM, but again its compiled C so it abuses the stack pointer and it too is hard to follow the ASM code. I need alot of help with it to be honest. 

But the machine was designed to download weather and graphics data from the satellite (at a cable headend) and display it on the TVs over the cable network at certain times.

**Comment by thetechknight** [comment]
Here is a link to the pictures of the PCBs that go in the unit. 

https://drive.google.com/open?id=1cHbiTzum0cIDc6h4PnnDj9PNHb7TR8KU
